### ISSUE TITLE
Letting VemTablineGo fail if passed number isn't a buffer index

### ISSUE THREAD

**Issue by noib3** [pacha/vem-tabline#20]
The documentation says that when using `let g:vem_tabline_show_number = 'none'` or `'buffnr'`, the `:buffer` command should be used for switching buffers, while `:VemTablineGo` should be used when using the `'index'` option.

However, `VemTabLineGo` is defined as follows

```vim
" User function to switch buffers
function! VemTablineGo(tagnr) abort
    try
        let buffnr = g:vem_tabline#buffers#section.tagnr_map[a:tagnr . g:vem_tabline_number_symbol]
        exec 'buffer' . buffnr
    catch /E716:/
        exec 'buffer ' . a:tagnr
    catch //
        echoerr "VemTabline: Buffer " . a:tagnr . " does not exist"
    endtry
endfunction
```
because of the `catch /E716:/`, if the number being passed is not one of the buffer indices, the function won't fail as long as there's a buffer whose number matches the passed one. By buffer index I mean the one displayed in the tabline when using the `'index'` option, and by buffer number I mean the one shown by `:ls`.

I think if the user wants to switch buffer by its number he/she should use `:buffer`, while `:VemTablineGo` should be reserved for switching buffers by index.

TL;DR

`E716` shouldn't be catched in `VemTablineGo`. If the number passed is not one of the indices, the function should be left free to fail.

**Comment by pacha** [comment]
Somehow I went the dumb way trying to be smart... You're on point there. Changed in 6b29cd3b1240bac006f15edd682fd98471cc4cf4 :slightly_smiling_face:
